BitWatch/HomeViewController.swift

```diff
@@ -10,84 +10,124 @@ import UIKit
-
-// Views
-var mainView: UIScrollView = UIScrollView();
-var homeView: UIView = UIView();
-var workView: UIView = UIView();
-var restView: UIView = UIView();
-
-// Labels and Buttons
-var beginButton: UIButton = UIButton();
-var countDownLabel: UILabel = UILabel();
-var countDownTimer: NSTimer = NSTimer();
-var countDown = 0;
-
-// Colors
-var intervalGreen: UIColor = UIColor();
-var workPurple: UIColor = UIColor();
-var restRed: UIColor = UIColor();
-
-// Text Fields
-var workSetting: UITextField = UITextField();
-var restSetting: UITextField = UITextField();
-var intervalSetting: UITextField = UITextField();
-var menuOpen: Bool = false;
-
 class HomeViewController: UIViewController, ADBannerViewDelegate {
-    
-    @IBOutlet var adBannerView: ADBannerView!
+    enum HomeViewControllerState: Int{
+        case home = 0
+        case work = 1
+        case rest = 2
+        case settings = 3
+    }
+    var state: HomeViewControllerState = .home
+    
+    var adBannerView: ADBannerView!
+    
+    var countDownLabel: UILabel!
+    var countDownTimer: NSTimer!
+    var countDown = 0;
+    
+    //Main Views
+    var containerView: ContainerView!
+    var homeView: HomeView!
+    var workView: WorkView!
+    var restView: RestView!
+    var settingsView: SettingsView!
+    
+    //Sizes
+    var height: CGFloat!
+    var width: CGFloat!
+    
+    // Colors
+    var intervalGreen: UIColor = UIColor().intervalGreen()
+    var workPurple: UIColor = UIColor().workPurple()
+    var restRed: UIColor = UIColor().restRed()
 
-    override func viewDidLoad() {
-        super.viewDidLoad()
-        
-        // Create the Main Scroll View View
-        height = self.view.frame.size.height;
-        width = self.view.frame.size.width;
-        
-        // iAd Stuff
+    func setupIAds(){
         self.canDisplayBannerAds = true;
         self.adBannerView = ADBannerView(frame: CGRectMake(0, height - 50, width, 50));
         self.adBannerView.delegate = self;
         self.adBannerView.hidden = true;
+    }
+    override init() {
+        super.init()
+    }
+    required init(coder aDecoder: NSCoder) {
+        super.init(coder:aDecoder)
+    }
+    override func viewDidLoad() {
+        super.viewDidLoad()
         
-        countDown = 3;
-        intervalGreen = UIColor(red:(124.0/255.0), green:(208.0/255.0), blue:(126.0/255.0), alpha: 1.0);
-        workPurple = UIColor(red: (178.0/255.0), green: (124.0/255.0), blue: (209.0/255.0), alpha: 1.0);
-        restRed = UIColor(red: (208.0/255.0), green: (93.0/255.0), blue: (93.0/255.0), alpha: 1.0);
-        
-        mainView = UIScrollView(frame: CGRectMake(0, 0, width, height - 50));
-        mainView.contentSize = CGSizeMake((width * 2.0) - 45.0, mainView.frame.size.height);
-        mainView.scrollEnabled = false;
-        mainView.pagingEnabled = false;
-        mainView.backgroundColor = UIColor.whiteColor();
-        
-        // Create the Home View
-        self.buildHomeView();
```

Comment:
// Create the Home View		
self.buildHomeView();


This is a good practice if you are writing code for people who aren't very familiar but it's obvious what self.buildHomeView() does so the comment becomes extraneous.